lib/db/mem.js

```diff
@@ -1042,6 +1055,28 @@ module.exports = function (log, error) {
     }
   }
 
+  Memory.prototype.accountRecord = function (emailBuffer) {
+    const normalizedEmail = emailBuffer.toString('utf8').toLowerCase()
+
+    if (! emails[normalizedEmail]) {
+      return P.reject(error.notFound())
+    }
+
+    const uid = emails[normalizedEmail].uid
+    return P.all([this.accountEmails(uid), this.account(uid)])
+      .spread((emails, account) => {
+
+        Object.keys(emails).some((key) => {
+          var emailRecord = emails[key]
+          if (emailRecord.uid.toString('hex') === uid.toString('hex') && emailRecord.isPrimary) {
+            account.primaryEmail = emailRecord.normalizedEmail
```

Comment:
This does modify the account object that is returned, however I think this is consistent with the [SP](https://github.com/mozilla/fxa-auth-db-mysql/blob/1430410d948ccaaa0b6b65bfc9a835b6f69542d3/lib/db/schema/patch-056-057.sql#L39), unless I am misunderstanding?